File: training-data/php/binary_gap.php
<?php
declare(strict_types=1);

function binaryGap(int $n): int
{
    $trimmed = rtrim(decbin($n), '0');
    return max(0, ...array_map('strlen', explode('1', $trimmed)));
}

foreach ([9, 529, 20, 15, 1041] as $n) echo $n, ' ', decbin($n), ' ', binaryGap($n), PHP_EOL;

File: training-data/python/binary_gap.py
def binary_gap(n: int) -> int:
    best, last = 0, None
    i = 0
    while n:
        if n & 1:
            if last is not None:
                best = max(best, i - last - 1)
            last = i
        n >>= 1
        i += 1
    return best


if __name__ == "__main__":
    for n in [9, 529, 20, 15, 1041]:
        print(n, bin(n), binary_gap(n))
